### PR TITLE
UIEH-1277: Jest+RTL > Test UsageConsolidationContentResource

### PR DESCRIPTION
**Pull request by Terala-Priyanka**
## Purpose

Adding tests for UsageConsolidationContentResource component using Jest+RTL
Issue: https://issues.folio.org/browse/UIEH-1277

### CODE REVIEW

File: src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  it('should render a div with message when isFailed of costPerUseData is true', () => {
+    const newCostPerUseData = {
+      ...costPerUseData,
+      isFailed: true
+    };
+    const { getByText } = renderUsageConsolidationContentResource({
+      costPerUseData: newCostPerUseData
+    });
+
+    expect(getByText('ui-eholdings.usageConsolidation.fullTextRequestUsageTable.noResponse'));
```
Comment (BogdanDenis) [inline src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js:76]: Will this fail if the element is not found? Please add `.toBeDefined()` to expect to be 100% sure

File: CHANGELOG.md [hunk 2]
```diff
@@ -429,3 +429,4 @@
 * Custom Embargo: Removing Custom Embargo will not enable Save button (UIEH-668)
 * Make eholdings UI permissions effective (UIEH-669)
 * Tags: assign/unassign tags on packages regardless of isSelected value (UIEH-670)
+* Jest+RTL > Test UsageConsolidationContentResource (UIEH-1277)
```
Comment (BogdanDenis) [inline CHANGELOG.md:432]: Looks like you accidentally added this to old version - we're adding newest version to the top of changelog, and older version are at the bottom
Current WIP version is 7.2.0

**Review by BogdanDenis — CHANGES_REQUESTED** [review]

File: src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            startMonth: '',
+        },
+        usage:{
+            platforms:[
+                {
+                    counts: [3, 6, 16],
+                    isPublisherPlatform: true,
+                    name: "name",
+                    total: 17
+                }
```
Comment (mariiaschv) [inline src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js:28]: When we have multiline objects or arrays, we should put trailing commas after the last item, so please add them here and in all other places of this file :)

```suggestion
                    total: 17,
                },
```

File: src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+import { costPerUseTypes } from '../../../constants';
+
+const costPerUseData = {
+  data: {
+    [costPerUseTypes.RESOURCE_COST_PER_USE]:{
+      attributes: {
+        analysis: {
+            cost: 1,
+            costPerUse: 1,
+            usage: 1
```
Comment (mariiaschv) [inline src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js:15]: The indentations should be 2 spaces and not 4, please check with eslint (for convenience you can add an extension to the IDE you're working in) and fix all these warnings.

File: CHANGELOG.md (hunk 2 above)
Comment (Terala-Priyanka) [reply CHANGELOG.md]: updated as per the comment.

File: src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    <UsageConsolidationContentResource
+      costPerUseData={costPerUseData}
+      {...props}
+    />
+  </Harness>
+);
+
+describe("Given UsageConsolidationContentResource", () => {
+
+  it('No DOMelement should be rendered when costPerUseData.data.resourceCostPerUse is null', () => {
```
Comment (mariiaschv) [inline src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js:51]: We use unit tests naming convention where `it` clause starts with "should...". And also there's no need to add an empty line here.

```suggestion
describe("Given UsageConsolidationContentResource", () => {
  it('should not render DOMelement when costPerUseData.data.resourceCostPerUse is null', () => {
```

File: src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+    expect(getByText('ui-eholdings.usageConsolidation.fullTextRequestUsageTable.noResponse')).toBeDefined();
+  });
+
+  it('should render NoCostPerUseAvailable when noCostPerUseAvailable and isPlatformsDataEmpty are true', () => {
+    const newCostPerUseData = {
+      ...costPerUseData,
+      data:{
+        ...costPerUseData.data,
+        [costPerUseTypes.RESOURCE_COST_PER_USE] : {
```
Comment (mariiaschv) [inline src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js:84]: Please don't forget to add a space before opening curly brackets, and on the contrary before the colon there's no need for a space.

```suggestion
      data: {
        ...costPerUseData.data,
        [costPerUseTypes.RESOURCE_COST_PER_USE]: {
```

File: src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+            },
+            usage: {
+                platforms: [],
+                totals: {}
+            }
+          }
+        }
+      }
+    };
+    const { getByText } = renderUsageConsolidationContentResource({
```
Comment (mariiaschv) [inline src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js:100]:
```suggestion
    };
    
    const { getByText } = renderUsageConsolidationContentResource({
```

File: src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+      costPerUseData: newCostPerUseData
+    });
+    
+    expect(getByText('ui-eholdings.usageConsolidation.summary.resource.noData')).toBeDefined();
+  });
+
+  it('render Summary Table and FullTextRequestUsageTable when costPerUseData hold appropriate values', () => {
+    const {getByText} = renderUsageConsolidationContentResource();
```
Comment (mariiaschv) [inline src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js:108]:
```suggestion
  it('should render Summary Table and FullTextRequestUsageTable when costPerUseData hold appropriate values', () => {
    const { getByText } = renderUsageConsolidationContentResource();
```

**Review by mariiaschv — CHANGES_REQUESTED** [review]

File: src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js (hunk 7 above)
Comment (Terala-Priyanka) [reply src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js]: Implemented the changes in latest commit

File: src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js (hunk 6 above)
Comment (Terala-Priyanka) [reply src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js]: Implemented the changes in latest commit

File: src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js (hunk 5 above)
Comment (Terala-Priyanka) [reply src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js]: Implemented the changes in latest commit

File: src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js (hunk 4 above)
Comment (Terala-Priyanka) [reply src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js]: Implemented the changes in latest commit

File: src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js (hunk 3 above)
Comment (Terala-Priyanka) [reply src/features/usage-consolidation-accordion/usage-consolidation-content-resource/usage-consolidation-content-resource.test.js]: Implemented the changes in latest commit

**Review by mariiaschv — APPROVED** [review]

**Review by Svitlana-Korol — APPROVED** [review]

**Review by BogdanDenis — APPROVED** [review]

**Pull request merged**